cylc/flow/hostuserutil.py

```diff
@@ -189,8 +189,10 @@ def is_remote_host(self, name):
                 except IOError:
                     self._remote_hosts[name] = True
                 else:
-                    self._remote_hosts[name] = (
-                        host_info != self._get_host_info())
+                    # look for common IP with localhost
+                    self._remote_hosts[name] = not bool(
+                        set(self._get_host_info()[2]).intersection(
+                            set(host_info[2])))
```

Comment:
Looks good. Can also use the `&` operator syntactic sugar:

```suggestion
                    self._remote_hosts[name] = not (
                        set(self._get_host_info()[2]) & set(host_info[2]))
                    )
```